### DISCUSSION TITLE
Insta-Crash debugging in Chrome

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#2686] [Q&A]
I noticed a few days ago that my debug sessions where instantly crashing.  That is to say, as soon as the Oqtane theme renders and the main page is shown the browser crashes.  In trying things, I found that switching to another browser would allow me to debug.  then other folks on my projects started seeing this as well.  We are running v3.2.1 of Oqtane and have not upgraded yet.  I'm wondering if others are seeing this problem and what solutions you might have.  Below is the output at time of crash. 

![image](https://user-images.githubusercontent.com/311063/225412863-5a8a2a28-c731-4eb9-9c9e-dfc3f060c0ef.png)

**Comment by leigh-pointer** [comment]
@markdav-is Image I am struggling with ! 
I have been using the new versions extensively, other than issues with permissions and caching I haven't had any crashing issues.  My debugging scenario is Oqtane.Server as statup in my VS solution.  I only ever build the module when in debug or release.

**Comment by leigh-pointer** [comment]
Since yesterday evening late! I have been getting this:
![Oqtane-Bomb](https://user-images.githubusercontent.com/9447612/225602074-9aa58c16-3abc-4c49-8a23-134e4b118b0c.gif)
![image](https://user-images.githubusercontent.com/9447612/225602360-b02f04e8-e028-4b5c-8547-99529df169fb.png)

@markdav-is do you think this is the same or related?

**Comment by leigh-pointer** [comment]
Further I have just downloaded the Epic browser and My solutions work fine.  What I do see though (in Epic) is when the framework is loads , it reloads.

**Comment by sbwalker** [comment]
I noticed this happens when running on Blazor Server sometimes - and it seemed to start happening when the auto reconnect logic was added to the framework. The auto reconnect logic is intended to identify when Blazor Server loses its SignalR connecttion and displays the reconnect modal at the top of the browser, and automatically reconnect to the Server so that the end user does not have to click the Reload link manually. The script is being used by many other Blazor applications - not only Oqtane:

```
<script>
    // Blazor Server Reconnect
    new MutationObserver((mutations, observer) => {
        if (document.querySelector('#components-reconnect-modal h5 a')) {
            async function attemptReload() {
                await fetch('');
                location.reload();
            }
            observer.disconnect();
            attemptReload();
            setInterval(attemptReload, 5000);
        }
    }).observe(document.body, { childList: true, subtree: true });
</script>
```

My suspicion is that during startup there is sometimes some latency before Blazor establishes the SignalR connection with the Server - and the auto reconnect is taking action too soon, resulting in the behavior you observe. 

In my environment when this happens, it refreshes a few times and then behaves as expected. In your environment are you saying that it stays in the this refresh loop indefinitely?

Note that I do not use Chrome - I don't even have it installed on my system. I use Firefox.

**Reply by leigh-pointer** [reply]
@sbwalker yes it goes in to a loop retrying then stops/crashes

**Reply by sbwalker** [reply]
and it happens consistently all of the time? Only in your local environment or in other environments as well?

**Reply by leigh-pointer** [reply]
@sbwalker all the time but! When running just the framework from VS no problem. When running / debugging modules then problem happens all the time.

**Comment by wcrisafu** [comment]
Hi,

I had the same issue yesterday where I had not touched any code for two weeks, I started VS and after a couple of runs (with no code change), it stopped working. Now, when I launch the solution (IISExpress or command prompt) and when I expect the login to show up it just shut down (no error, no message).  My code adds an additional SignalR for real time messages (not chat) but I never encountered any issues till yesterday. Maybe some MS update is causing this?

**Comment by wcrisafu** [comment]
I was searching on google and saw this article. Is it related?
https://learn.microsoft.com/en-us/answers/questions/1189857/blazor-app-crashes-when-debugging-after-windows-se

**Comment by markdav-is** [comment]
this seemed toe clear itself up after a few days, so I suspect it was a Chrome update that was quickly patched.
